kong/tools/timestamp.lua

```diff
@@ -40,31 +40,46 @@ local function get_timetable(now)
   return tt_from_timestamp(timestamp)
 end
 
+
 --- Creates a timestamp table containing time by different precision levels.
 -- @param now (optional) Time to generate timestamps from, if omitted current UTC time will be used
+-- @param tracked_timestamp (optional) Timestamp tracked, pairs of key/value in : {["second"]= true, ["minute"]= true, ["hour"]= true, ["day"]= true, ["month"]= true, ["year"]= true}
 -- @return Timestamp table containing fields/precisions; second, minute, hour, day, month, year
-local function get_timestamps(now)
+local function get_timestamps(now, tracked_timestamp)
   local timetable = get_timetable(now)
+  local tracked_timestamp = tracked_timestamp or {["second"]= true, ["minute"]= true, ["hour"]= true, ["day"]= true, ["month"]= true, ["year"]= true}
```

Comment:
This will incur in one table allocation for every call. Please initialize said table once at the beginning and reuse it instead:

```suggestion
  local tracked_timestamp = tracked_timestamp or DEFAULT_TIMESTAMPS
```